jsk_apc2016_common/python/jsk_apc2016_common/__init__.py

```diff
@@ -5,13 +5,127 @@
+OLD_PKG_PATH = rp.get_path(OLD_PKG)
 
 def get_object_data():
     fname = osp.join(rp.get_path(PKG), 'data/object_data.yaml')
     data = yaml.load(open(fname))
     return data
+
+def load_stow_json(json_file):
+    json_data = json.load(open(json_file))
+    bin_contents = {}
+    for bin, item in json_data['bin_contents'].items():
+        bin = bin[len('bin_'):].lower()
+        bin_contents[bin] = item
+    work_order = []
+    for contents in json_data['work_order']:
+        work_order.append(contents)
+    return bin_contents, work_order
+
+def visualize_stow_json(json_file):
+    """Visualize bin_contents and work_order with a json file"""
+    from jsk_apc2015_common.util import rescale
+    # load data from json
+    bin_contents, work_order = load_stow_json(json_file)
+    # initialize variables
+    kiva_pod_img = cv2.imread(osp.join(OLD_PKG_PATH, 'models/kiva_pod/image.jpg'))
+    tote_img = cv2.imread(osp.join(PKG_PATH, 'models/tote/image.jpg'))
+
+    BIN_REGION = {
+        'a': ((0, 50), (640, 610)),
+        'b': ((640, 50), (1410, 610)),
+        'c': ((1410, 50), (2060, 610)),
+        'd': ((0, 680), (640, 1200)),
+        'e': ((640, 680), (1410, 1200)),
+        'f': ((1410, 680), (2060, 1200)),
+        'g': ((0, 1280), (640, 1770)),
+        'h': ((640, 1280), (1410, 1770)),
+        'i': ((1410, 1280), (2060, 1770)),
+        'j': ((0, 1850), (640, 2430)),
+        'k': ((640, 1850), (1410, 2430)),
+        'l': ((1410, 1850), (2060, 2430)),
+    }
+    # get object images
+    object_list = get_object_data()
+    object_imgs = {}
+    for obj in object_list:
+        obj = obj['name']
+        img_path = osp.join(OLD_PKG_PATH, 'models/{obj}/image.jpg'.format(obj=obj))
+        img = cv2.imread(img_path)
+        h, w = img.shape[:2]
+        if h > w:
+            img = np.rollaxis(img, 1)
+        object_imgs[obj] = img
+    # draw objects
+    for bin, contents in bin_contents.items():
+        bin_pt1, bin_pt2 = BIN_REGION[bin]
+        bin_region = kiva_pod_img[bin_pt1[1]:bin_pt2[1], bin_pt1[0]:bin_pt2[0]]
+        x_num, y_num = jsk_apc2015_common._get_tile_shape(len(contents))
+        bin_h, bin_w = bin_region.shape[:2]
+        max_obj_h, max_obj_w = bin_h // y_num, bin_w // x_num
+        for i_y in xrange(y_num):
+            y_min = int(1. * bin_h / y_num * i_y)
+            for i_x in xrange(x_num):
+                x_min = int(1. * bin_w / x_num * i_x)
+                if contents:
+                    obj = contents.pop()
+                    obj_img = object_imgs[obj]
+                    scale_h = 1. * max_obj_h / obj_img.shape[0]
+                    scale_w = 1. * max_obj_w / obj_img.shape[1]
+                    scale = min([scale_h, scale_w])
+                    obj_img = rescale(obj_img, scale)
+                    obj_h, obj_w = obj_img.shape[:2]
+                    x_max, y_max = x_min + obj_w, y_min + obj_h
+                    bin_region[y_min:y_max, x_min:x_max] = obj_img
```

Comment:
`jsk_apc2015_common.visualize_json` needs to be edited for a few lines to be reused. That is why I was thinking not to reuse :). I will edit those.